Method reply support on Windows and automated tests

vjrantal opened the pull request.

vjrantal committed:
Added tests for replying to methods.

Now that we have more test code, split the tests into separate files
per the functional area they are testing.

vjrantal committed:
Allow calling connect multiple times.

This is useful, for example, in automated tests so that we can
safely connect from multiple independent test files regardless
of if connection was created already elsewhere.

vjrantal commented on tests/methods.js:
```diff
@@ -0,0 +1,88 @@
+                    }, function (status) {
+                        errorCallback(status);
+                    }, session.sessionHost,
+                    // TODO: Passing '/path' as parameter instead of null results
```
@obsoleted Do you happen to know why I wasn't able to pass '/path' to the `callMethod`? I can try to provide more detailed error trace if that helps.

obsoleted replied:
I do not. I'll try to repro and investigate on my side also.

vjrantal committed:
Method reply support for Windows.

The AllJoyn message loop had some corner cases where messages weren't properly
closed and this commit fixes those issues.

Exposed the error member from the message struct to JavaScript so that it
can be returned in case AllJoyn messages are marked as error messages.

vjrantal committed:
Don't prefix the error message on iOS.
